sourcegraph/learn
# Add dates

### What should this PR do?
[DEVED-147](https://sourcegraph.atlassian.net/jira/software/projects/DEVED/boards/9?selectedIssue=DEVED-147)
Add dates to all posts, "Published on `$DATE` • Updated on `$DATE`"

Include flow control for if there is not an updated date.

### Why are we making this change?
It will help us with reporting (knowing when items are published and knowing if items get out of date).

It will help readers know if something was recently published so that they can determine whether it is up to date enough for them to read.

### What are the acceptance criteria? 
Ensure that everything is working as expected across posts.

Also, style feedback would be helpful. I changed the order of items, and put the dates in as US format (following the approach on the [Sourcegraph blog](https://about.sourcegraph.com/blog/).

Screenshots:

<img width="774" alt="Screen Shot 2021-09-03 at 11 25 59 AM" src="https://user-images.githubusercontent.com/5977693/132031199-d3a522b1-82b2-4847-a160-e61cbc481846.png">

<img width="765" alt="Screen Shot 2021-09-03 at 11 26 09 AM" src="https://user-images.githubusercontent.com/5977693/132031207-7a12c0b2-0813-46bf-9601-595fae8663a1.png">

### How should this PR be tested?
Check out the branch or review the deploy preview and make sure that the frontend is working as expected.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
